File: benchmark/perl/bubble.pl
use Time::HiRes qw(time);
my $n = 500;
my @arr = reverse 1..$n;
my $start = time();
for my $k (0..$n-1) {
    for my $j (0..$n-2) {
        if ($arr[$j] > $arr[$j + 1]) {
            @arr[$j, $j+1] = @arr[$j+1, $j];
        }
    }
}
my $end = time();
my $ms = ($end - $start) * 1000;
print "First: $arr[0]
";
print "Last: $arr[$n-1]
";
printf "Time: %.0fms
", $ms;

File: benchmark/php/bubble.php
<?php
$n = 500;
$arr = [];
for ($i = $n; $i > 0; $i--) { $arr[] = $i; }
$start = hrtime(true);
for ($k = 0; $k < $n; $k++) {
    for ($j = 0; $j < $n - 1; $j++) {
        if ($arr[$j] > $arr[$j + 1]) {
            $t = $arr[$j]; $arr[$j] = $arr[$j + 1]; $arr[$j + 1] = $t;
        }
    }
}
$end = hrtime(true);
$ms = ($end - $start) / 1e6;
echo "First: " . $arr[0] . "
";
echo "Last: " . $arr[$n - 1] . "
";
echo "Time: " . round($ms) . "ms
";
